airbyte-integrations/connectors/source-square/source_square/components.py

```diff
@@ -0,0 +1,142 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+import logging
+from dataclasses import InitVar, dataclass
+from typing import Any, ClassVar, Iterable, Mapping, MutableMapping, Optional, Union
+
+import pendulum
+from dataclasses_jsonschema import JsonSchemaMixin
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.auth import DeclarativeOauth2Authenticator
+from airbyte_cdk.sources.declarative.auth.declarative_authenticator import DeclarativeAuthenticator
+from airbyte_cdk.sources.declarative.auth.token import BearerAuthenticator
+from airbyte_cdk.sources.declarative.stream_slicers import StreamSlicer
+from airbyte_cdk.sources.declarative.types import Record, StreamSlice, StreamState
+from airbyte_cdk.sources.streams.core import Stream
+
+
+@dataclass
+class AuthenticatorSquare(DeclarativeAuthenticator, JsonSchemaMixin):
+    config: Mapping[str, Any]
+    bearer: BearerAuthenticator
+    oauth: DeclarativeOauth2Authenticator
+
+    def __new__(cls, bearer, oauth, config, *args, **kwargs):
+        if config.get('api_key'):
+            return bearer
+        else:
+            return oauth
+
+
+@dataclass
```

Comment:
am I correct in saying that the point of this slicer is to handle just an additional date format, and should therefore be in the CDK itself? 

If so, could you: 
1. Create an issue explaining the functionality needed
2. Link this issue in the docstring of this class
3. As with above, I'd love to see a suggestion for how to incorporate this into the lowcode language. Once we reach alignment we can look at a PR!

Reply:
Explained in https://github.com/airbytehq/airbyte/issues/19760